tools/mksurfdata_map/src/mkgridmapMod.F90

```diff
@@ -688,21 +688,19 @@ subroutine gridmap_areaave_srcmask (gridmap, src_array, dst_array, nodata, mask_
 ! !LOCAL VARIABLES:
     integer :: n,ns,ni,no
     real(r8):: wt
-    real(r8), allocatable :: wtnorm(:)
     character(*),parameter :: subName = '(gridmap_areaave_srcmask) '
 !EOP
 !------------------------------------------------------------------------------
     call gridmap_checkifset( gridmap, subname )
     ns = size(dst_array)
-    allocate(wtnorm(ns)) 
-    wtnorm(:) = 0._r8
+    gridmap%frac_dst(:) = 0._r8
```

Comment:
I think I understand why you changed wtnorm to just use gridmap%frac_dst, but I feel this muddies the interface of this routine: The code flow is cleaner when everything in gridmap is set in one place (before calling this routine) and then it's an intent(in) argument to this routine. (The compiler doesn't complain about you setting frac_dst on the intent(in) gridmap because Fortran lets you change values of pointer components of an intent(in) object, but I find this really confusing.)

So I'd much prefer if you went back to something like before, but wtnorm can be a new intent(out) argument that is passed back to the caller.

Reply:
Will do.